src/Hosting/AspNetCore/Http/PrincipalProvider.cs

```diff
@@ -13,5 +13,10 @@ public PrincipalProvider(IHttpContextAccessor accessor)
             this.accessor = accessor;
         }
         public IPrincipal Current => this.accessor.HttpContext.User;
```

Comment:
We should also do a null check. If someone forgot to check the "HasValue" this will raise the exception
```
public IPrincipal Current => this.accessor.HttpContext?.User;
```